setup.py

```diff
@@ -4,53 +4,58 @@
   from setuptools import setup, Extension
 except ImportError:
   from distutils.core import setup, Extension
-import sys, importlib, os, glob, io
+import importlib
+import glob
+import io
+
 
 def version():
   loader = importlib.machinery.SourceFileLoader("hiredis.version", "hiredis/version.py")
   module = loader.load_module()
   return module.__version__
 
+
 ext = Extension("hiredis.hiredis",
-  sources=sorted(glob.glob("src/*.c") +
-                 ["vendor/hiredis/%s.c" % src for src in ("alloc", "read", "sds")]),
-  extra_compile_args=["-std=c99"],
-  include_dirs=["vendor"])
+                sources=sorted(glob.glob("src/*.c") +
+                               ["vendor/hiredis/%s.c" % src for src in ("alloc", "async", "hiredis", "net", "read", "sds")]),
+                extra_compile_args=["-std=c99"],
+                extra_link_args=["-Wl,-Bsymbolic"],
```

Comment:
-Bsymbolic fails across many platforms - please see CI

Reply:
@chayim @dvora-h 
Removed.
It crashes in RedisGears without the extra link args. Though we can figure this out in a different PR.

Reply:
I suggest we leave it in there except for the MacOS or platforms that don't support the flag. Or vice versa, but this conflicts with Redis' hiredis and crashes in RE without this

Reply:
Bsymbolic is a pretty large hammer to wield. Given that we precompile this and push various wheels to pypi, I'd like to be careful with this outcome, especially as we're now changing everyone.

I think you should be safe with the following ```return ["-std=c99", "-static-libstdc++", "-O2"]``` in place of the current ```-stc=c99``` should acheive the same effect. Note: we did add a compiler optimization there, and one could argue removing and -O3 as well (which I'm open to).

@prokazov what happens with gears (and others) in this case? Symbol collisions or sanity?